app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetail2Fragment.kt

```diff
@@ -0,0 +1,355 @@
+
+    private val topicsObserver = Observer<PagedList<DiscussionTopicListDetail>?> {
+            t -> topicsRecyclerAdapter?.submitList(t)
+    }
+
+
+
+    override var topics: DataSourceFactory<Int, DiscussionTopicListDetail>? = null
+        set(value) {
+
+            topicsLiveData?.removeObserver(topicsObserver)
+            field = value
+            val topicsDao = repo?.discussionTopicDao ?:return
+            topicsLiveData = value?.asRepositoryLiveData(topicsDao)
+            topicsLiveData?.observe(this, topicsObserver)
+        }
+
+
+
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val rootView: View
+
+        mBinding = FragmentCourseDiscussionDetailBinding.inflate(inflater, container,
+            false).also {
+            rootView = it.root
+            it.fragmentCourseDiscussionDetailRv
+
+        }
+
+        detailMergerRecyclerView =
+            rootView.findViewById(R.id.fragment_course_discussion_detail_rv)
+
+        // 1
+        descriptionRecyclerAdapter = CourseDiscussionDescriptionRecyclerAdapter()
+
+        // 2
+        topicsHeadingRecyclerAdapter = SimpleHeadingRecyclerAdapter(getString(R.string.topics))
+        topicsHeadingRecyclerAdapter?.visible = true
+
+
+
+
+        val accountManager: UstadAccountManager by instance()
+        repo = di.direct.on(accountManager.activeAccount).instance(tag = DoorTag.TAG_REPO)
+        mPresenter = CourseDiscussionDetailPresenter(requireContext(), arguments.toStringMap(), this,
+            di, viewLifecycleOwner).withViewLifecycle()
+        mPresenter?.onCreate(savedInstanceState.toNullableStringMap())
+
+        // 3
+        topicsRecyclerAdapter = DiscussionTopicRecyclerAdapter(mPresenter)
+
+
+        detailMergerRecyclerAdapter = ConcatAdapter(descriptionRecyclerAdapter,
+            topicsHeadingRecyclerAdapter, topicsRecyclerAdapter)
+
+        detailMergerRecyclerView?.adapter = detailMergerRecyclerAdapter
+        detailMergerRecyclerView?.layoutManager = LinearLayoutManager(requireContext())
+
+
+
+        return rootView
+    }
+
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mBinding = null
+        mPresenter = null
+        entity = null
+
+        descriptionRecyclerAdapter = null
+        topicsHeadingRecyclerAdapter = null
+        detailMergerRecyclerView?.adapter = null
+        detailMergerRecyclerView = null
+
+
+
+    }
+
+
+    override var entity: CourseDiscussion? = null
+        set(value) {
+            field = value
+            descriptionRecyclerAdapter?.courseDiscussion = value
+            ustadFragmentTitle = value?.courseDiscussionTitle
+        }
+
+
+
+}
+
+
+
+@Composable
+private fun PostsDetail(
+    posts: List<DiscussionPostWithDetails> = emptyList(),
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {}
+){
+
+    LazyColumn(
```

Comment:
Nested LazyColumn won't work (its the same as nesting a recyclerview). It needs to be the top level container.